### ISSUE TITLE
I integrate html5 with iOS webview,close app do not trigger disconnect and raise Exception!

### ISSUE THREAD

**Issue by larry-dev** [miguelgrinberg/python-socketio#107]
i use:
aiohttp 2.1.0
python-engineio 1.5.4
python-socket.io 1.7.5

raise Exception:

```
Traceback (most recent call last):
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/engineio/asyncio_socket.py", line 180, in _websocket_handler
    p = await asyncio.wait_for(wait_task, self.server.ping_timeout)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/asyncio/tasks.py", line 345, in wait_for
    yield from waiter
concurrent.futures._base.CancelledError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/aiohttp/web_protocol.py", line 421, in start
    resp = yield from self._request_handler(request)
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/aiohttp/web.py", line 303, in _handle
    resp = yield from handler(request)
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/engineio/asyncio_server.py", line 137, in handle_request
    packets = await socket.handle_get_request(environ)
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/engineio/asyncio_socket.py", line 73, in handle_get_request
    return await getattr(self, '_upgrade_' + transport)(environ)
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/engineio/asyncio_socket.py", line 120, in _upgrade_websocket
    return await ws(environ)
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/engineio/async_aiohttp.py", line 95, in __call__
    await self.handler(self)
  File "/Volumes/macsource/01-Project/02-Python/yikeweiqi/venv/lib/python3.6/site-packages/engineio/asyncio_socket.py", line 187, in _websocket_handler
    wait_task.exception()
asyncio.base_futures.InvalidStateError: Result is not ready.
```

**Comment by larry-dev** [comment]
sorry,it is my problem.

**Comment by miguelgrinberg** [comment]
There is a bug in asyncio that I think is causing this. The `Result is not ready` error comes as a result of some code that I added that I thought would workaround this asyncio bug. See https://github.com/miguelgrinberg/python-engineio/blob/master/engineio/asyncio_socket.py#L183-L188 for a bit of background on the error plus a link to the asyncio bug I filed. I tested this workaround on 3.6.0. Which version of Python are you using?

The upstream bug is going to be fixed in Python 3.6.2, which should be coming out in about 10-15 days. Would you be okay waiting for 3.6.2 to retest? If

**Comment by larry-dev** [comment]
thanks,i used python3.6.1
I'll be waiting for 3.6.2 to retest.
